substrate/client/transaction-pool/src/fork_aware_txpool/fork_aware_txpool.rs

```diff
@@ -812,21 +825,21 @@ where
 		);
 		let xt = Arc::from(xt);
 
-		let InsertionInfo { hash: xt_hash, source: timed_source, .. } =
-			match self.mempool.push_watched(source, xt.clone()) {
-				Ok(result) => result,
-				Err(TxPoolApiError::ImmediatelyDropped) =>
-					self.attempt_transaction_replacement(source, true, xt.clone()).await?,
-				Err(e) => return Err(e.into()),
-			};
+		let insertion = match self.mempool.push_watched(source, xt.clone()) {
+			Ok(result) => result,
+			Err(TxPoolApiError::ImmediatelyDropped) =>
```

Comment:
nit(non-topic/might be done in a follow up): wdyt about renaming this error to be more suggestive about the fact that it is raised when the limits are hit?

Reply:
The name was there for a while, for me it sounds pretty nice, also has good documentation.